tests/tests/clear_texture.rs

```diff
@@ -267,13 +267,21 @@ async fn single_texture_clear_test(
 async fn clear_texture_tests(ctx: TestingContext, formats: &'static [wgpu::TextureFormat]) {
     for &format in formats {
         let (block_width, block_height) = format.block_dimensions();
-        let rounded_width = block_width * wgpu::COPY_BYTES_PER_ROW_ALIGNMENT;
-        let rounded_height = block_height * wgpu::COPY_BYTES_PER_ROW_ALIGNMENT;
+        let rounded_width = if block_width == 1 {
+            wgpu::COPY_BYTES_PER_ROW_ALIGNMENT
+        } else {
+            block_width
+        };
+        let rounded_height = if block_height == 1 {
+            wgpu::COPY_BYTES_PER_ROW_ALIGNMENT
+        } else {
+            block_height
+        };
```

Comment:
I think this was correct previously.

Looking at the initial CI failure we need to bump the `max_texture_dimension_3d` to at least 1024.

Reply:
@teoxoy thank you for the feedback! Is that limit to be increased in the test device configuration or somewhere else?

Reply:
Yup, in the test device configuration. For example we bump up `max_texture_dimension_2d` for the ASTC test.

Reply:
Tried but ASTC block_width 12 bumps up the size too much (in general, including DX12, 3D Textures are limited to 2048 per dimension but 256*12 results in 3072), probably casing is necessary or least common multiple calculation

Reply:
Actually I don't know why we multiply by `wgpu::COPY_BYTES_PER_ROW_ALIGNMENT`. The validation of create texture only says the `width`/`height` should be a multiple of `block_width`/`block_height`.

Reply:
It seems our D3D12 backend can't handle some clear texture cases https://github.com/gfx-rs/wgpu/actions/runs/9278927977/job/25531020234. `writeTexture` also probably can't either.

Reply:
@teoxoy The D3D12 tests worked with this commit: https://github.com/gfx-rs/wgpu/actions/runs/9278142386/job/25528960851

We'll update the spec to make 3D textures only work with BCn initially, then maybe this PR will be simpler because we won't need to think about the case of ASTC.

Reply:
This is most likely https://github.com/gfx-rs/wgpu/issues/4150.

So I think we can avoid it for D3D12 devices specifically by only multiplying the block sizes with `wgpu::COPY_BYTES_PER_ROW_ALIGNMENT`.
And request a `max_texture_dimension_3d` of 1024.

Reply:
Sounds good, I'll try that

Reply:
@teoxoy seems like it works! But there's a suite of errors from macOS runner which I'm not getting on my local M1 machine, though I get one issue with Apple Silicon that's not related to compressed formats:


```
--- STDERR:              wgpu-test::wgpu-test [Executed Failure: BACKEND] [Vulkan/Apple M1/0] wgpu_test::shader::struct_layout::uniform_input ---
[2024-05-30T19:27:47Z ERROR wgpu_test::expectations] Expected to fail due to [FailureReason { kind: Some(ValidationError), message: Some("a matrix with stride 8 not satisfying alignment to 16") }], but did not fail
```

Reply:
I reran CI, the error you are seeing on your side is most likely because you don't have the vulkan validation layers.

Reply:
@teoxoy now there's failure from Linux runner in CI but that's not related to this PR, is my understanding correct?

```
  FAIL [   0.347s] wgpu-examples [Executed] [Gl/llvmpipe (LLVM 13.0.1, 256 bits)/1] wgpu_examples::timestamp_queries::tests::timestamps_passes
```

Reply:
That test is failing intermittently.